app/models/course/registration.rb

```diff
@@ -0,0 +1,29 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: roles.values_at(:teaching_assistant, :manager, :owner)) }
```

Comment:
But in the `CourseUser.roles` hash. I'm not sure if you're allowed to modify the enumeration.